pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -12,6 +12,7 @@ service ChainReader {
   rpc BatchGetLatestValues (BatchGetLatestValuesRequest) returns (BatchGetLatestValuesReply) {}
   rpc QueryKey(QueryKeyRequest) returns (QueryKeyReply) {}
   rpc Bind(BindRequest) returns (google.protobuf.Empty) {}
+  rpc ReplaySequence(ReplaySequenceRequest) returns (google.protobuf.Empty) {}
```

Comment:
what's the control flow and data flow here? it doesn't return anything. so something else is going to consume the result? i would have expected some sort of handle or stream as the return value so the caller can consume it.

as a corollary - what happens if this called concurrently? it doesn't appear to be safe to do so because there is no explicit handle to the result of the operation.

/nit, /dumbquestion: is there something other than a sequence that can be replayed?

Reply:
It returns an error, this is how Bind was handled also. Is there another way to define error return value through proto?
Yes, only sequences can be replayed